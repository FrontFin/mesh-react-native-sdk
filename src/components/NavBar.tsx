import { useSDKStyles } from './SDKContainer.styled';
import { Image, TouchableOpacity, View } from 'react-native';

type NavBarProps = {
  goBack: () => void;
  showCloseAlert: () => void;
}

export const NavBar = ({goBack, showCloseAlert}: NavBarProps) => {
  const {navBarStyles: styles} = useSDKStyles();

  return (
    <View testID={'native-navbar'} style={styles.navBarContainer}>
      <TouchableOpacity
        testID={'nav-back-button'}
        onPress={goBack} style={styles.navBarImgContainer}>
        <Image
          source={require('../assets/ic_back.png')}
          style={styles.navBarImgButton}
        />
      </TouchableOpacity>

      <TouchableOpacity
        onPress={showCloseAlert}
        style={styles.navBarImgContainer}
        testID={'close-button'}
      >
        <Image
          source={require('../assets/ic_close.png')}
          style={styles.navBarImgButton}
        />
      </TouchableOpacity>
    </View>
  );
}
