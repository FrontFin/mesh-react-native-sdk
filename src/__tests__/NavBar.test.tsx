import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import NavBar from '../components/NavBar';

describe('NavBar', () => {
  test('renders correctly', () => {
    const component = render(<NavBar goBack={() => void 0} showCloseAlert={() => void 0} />);

    const { getByTestId } = component;

    expect(getByTestId('native-navbar')).toBeTruthy();
    expect(getByTestId('nav-back-button')).toBeTruthy();
    expect(getByTestId('close-button')).toBeTruthy();

    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('calls goBack when the back button is pressed', () => {
    const goBackMock = jest.fn();
    const { getByTestId } = render(<NavBar goBack={goBackMock} showCloseAlert={() => {
      void 0
    }} />);

    fireEvent.press(getByTestId('nav-back-button'));

    expect(goBackMock).toHaveBeenCalledTimes(1);
  });

  test('calls showCloseAlert when the close button is pressed', () => {
    const showCloseAlertMock = jest.fn();
    const { getByTestId } = render(<NavBar goBack={() => {
      void 0
    }} showCloseAlert={showCloseAlertMock} />);

    fireEvent.press(getByTestId('close-button'));

    expect(showCloseAlertMock).toHaveBeenCalledTimes(1);
  });
});
