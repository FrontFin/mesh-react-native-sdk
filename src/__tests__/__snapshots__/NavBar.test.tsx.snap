// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "space-between",
      "paddingEnd": 8,
      "paddingStart": 6,
      "paddingTop": 12,
    }
  }
  testID="native-navbar"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": 40,
      }
    }
    testID="nav-back-button"
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/ic_back.png",
        }
      }
      style={
        {
          "height": 24,
          "resizeMode": "contain",
          "width": 24,
        }
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": 40,
      }
    }
    testID="close-button"
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/ic_close.png",
        }
      }
      style={
        {
          "height": 24,
          "resizeMode": "contain",
          "width": 24,
        }
      }
    />
  </View>
</View>
`;
